File: ABC/abc024/abc024_b 3.py
#100点解法
max_t = 1100001
open = [0] * max_t #いもす法

N, T = map(int, input().split())
for _ in range(N):
    A = int(input())
    open[A] += 1
    open[A+T] -= 1

for t in range(max_t - 1):
    open[t+1] += open[t]

print(sum(open[t] > 0 for t in range(max_t)))

File: ABC/abc024/abc024_b 3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 100点解法
    int max_t = 1100001, N, T, A, ans = 0;
    vector<int> open(max_t, 0); // いもす法

    cin >> N >> T;
    for (int i = 0; i < N; i++) {
        cin >> A;
        open[A]++;
        open[A + T]--;
    }

    for (int t = 0; t < max_t - 1; t++) {
        open[t + 1] += open[t];
        if (open[t] > 0) ans++;
    }
    cout << ans << endl;
}